apps/pipeline/lib/pipeline/writer/s3_writer/compaction.ex

```diff
@@ -81,6 +82,7 @@ defmodule Pipeline.Writer.S3Writer.Compaction do
     orc_table = options[:orc_table]
     json_table = options[:json_table]
     compact_table = "#{orc_table}_compact"
+    TelemetryEventHelper.add_dataset_record_event_count(orc_table, new_count)
```

Comment:
I'd suggest doing this last in this function so that it won't record numbers for a failed compaction